### PR TITLE
10 articles by topic

### PR DESCRIPTION
**Pull request by addleton**

### CODE REVIEW

File: src/utils/utils.js [hunk 1]
```diff
@@ -5,3 +5,9 @@ export const convertTime = (timestamp) => {
   const day = dateObject.getUTCDate();
   return `${day}-${month}-${year}`;
 };
+
+export const getTopicArticles = (articles, topic) => {
+  return articles.filter((article) => {
+    return article.topic === topic;
+  });
+};
```
Comment (itskatherine) [inline src/utils/utils.js:13]: I think this is a good approach given the problems you were having with the query 👍

File: src/components/Articles.jsx [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 import ArticleCard from "./ArticleCard";
 import LinearProgress from "@mui/joy/LinearProgress";
+import { useLocation } from "react-router-dom";
+import { getTopicArticles } from "../utils/utils";
 
 const Articles = () => {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
```
Comment (itskatherine) [inline src/components/Articles.jsx:12]: Cool use of URLSearchParams 👍

File: src/App.css [hunk 3]
```diff
@@ -20,6 +20,17 @@ nav {
...
+  justify-content: space-evenly;
+}
+
+.nav-button {
+  background-color: black;
+  color: white;
+  border-radius: 5px;
+  margin-top: 5px;
+  margin-bottom: 5px;
+  font-family: "Montserrat", sans-serif;
```
Comment (itskatherine) [inline src/App.css:33]: Love the styling btw 🎨

File: src/components/Articles.jsx [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+  const paramValue = query.get("topic");
 
   useEffect(() => {
     getAllArticles().then(({ data: { articles } }) => {
       setArticles(articles);
+      if (paramValue) {
+        const newArticles = getTopicArticles(articles, paramValue);
+        setArticles(newArticles);
```
Comment (itskatherine) [inline src/components/Articles.jsx:20]: Nice reuse of the Articles component

**Review by itskatherine — COMMENTED** [review]
Looks great Michael, good to merge 👍

**Comment by addleton** [comment]
> Looks great Michael, good to merge 👍

Thanks Katherine! :)

**Pull request merged**